agent/pagebroker/daemon_test.cpp

```
@@ -102,6 +177,135 @@ TEST_F(BrokerTest, RejectsSymlinkInRestoreSource)
+      invalid.mutable_staged_restore()->mutable_source(), invalid.mutable_staged_restore()->mutable_io_engine(),
+      "relative");
+  EXPECT_EQ(broker().HandleRequest(invalid).failure().code(), Failure::INVALID_REQUEST);
+
+  auto restore = RequestFor("restore");
+  Configure(
+      restore.mutable_staged_restore()->mutable_source(), restore.mutable_staged_restore()->mutable_io_engine(),
+      source_);
+  EXPECT_TRUE(broker().HandleRequest(restore).has_staged_restore_directory());
+}
+
+TEST_F(BrokerTest, RejectsPathsOutsideStorageRoot)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)